### ISSUE TITLE
Allow `#[Object]` resolver methods to return any error that implements `Into<async_graphql::Error>`

### ISSUE THREAD

**Issue by andybarron** [async-graphql/async-graphql#476]
## Description of the feature
I want to use a custom `Error` type for all queries so that they are guaranteed to have the same extensions. Right now, only `Result<T, async_graphql::Error>` (aliased to `async_graphql::Result<T>`) is allowed in the return position of resolver methods. I'm currently using a workaround - the macro will actually accept any time that implements `into_server_error`. (🦆 typing!)

## Code example (if possible)
```rust
pub struct Mutation;

#[Object]
impl Mutation {
    // Currently only works if SomeConvertibleError has a method called into_server_error()
    // that matches the declaration for async_graphql::Error.
    async fn do_something(&self) -> Result<Foo, SomeConvertibleError> {
        todo!()
    }
}
```

**Comment by sunli829** [comment]
This problem is not easy to solve, I will find a way.

**Comment by sunli829** [comment]
Released in `v2.8.2`

**Comment by andybarron** [comment]
amazing :) thank you!!!
